base/uk.ac.stfc.isis.ibex.ui.logplotter/src/uk/ac/stfc/isis/ibex/ui/logplotter/LogPlotterHistoryPresenter.java

```diff
@@ -113,13 +126,79 @@ private void addPVToEditor(String pvAddress, final String displayName, DataBrows
 			// Add item to new axes
 			item.setAxis(axis);
 			model.addItem(item);
+			
+			model.addListener(new ModelListenerAdapter() {
+			    private int dataCount = 0;
+			    private static final int DATA_COUNT_LIMIT = 10;
+			    private boolean autoscale = true;
+
+			    // This listener is triggered every time that a data point is added to the graph,
+                // either from archive data or from new data. After 10 pieces of data are received,
+			    // it turns off autoscale.
+	            @Override
+	            public void selectedSamplesChanged() {
+	                if (dataCount < DATA_COUNT_LIMIT) {
+	                    dataCount++;
+	                } else if (autoscale) {
+	                    Display.getDefault().asyncExec(new Runnable() {
```

Comment:
There's some slightly leaner syntax you can use here: `Display.getDefault().asyncExec(() -> axis.setAutoScale(false));`